### DISCUSSION TITLE
How can HALLO and Orcasound agree upon a nomenclature for broadband impulsive sounds made by SRKWs?

### DISCUSSION THREAD

**Discussion by oliskir** [orcasound/signals-srkw#16] [General]
The current (Sep 2024) iteration of the HALLO annotation taxonomy, employes the following hierarchical labelling structure for killer whale vocalisations:
```
├── BZ [Buzz]
├── CK [Echolocation click]
├── Rasp [Rasp]
├── SI [Surface Impact]
│   ├── J [Jumping]
│   └── TS [Tail Slap]
└── TC [Tonal Call]
    ├── PC [Pulsed Call]
    │   └── BP [Burst Pulse]
    └── W [Whistle]
```
With additional stereotyped calls (S01, S02, ... etc) nested under PC, but not shown in the diagram. 

What are people's thoughts on this taxonomy and its nested structure, especially in relation to the broadband impulsive sounds used by SRKWs? That is, CK, Rasp, BZ, and BP?

The current "definitions" we have in HALLO for these four sound types are:

## Rasp
Very fast series of modulated clicks used by the whale to gain higher precision, and it may often be associated with foraging. Used by Bigg's killer whales and possibly also by other ecotypes.  You can still hear/see the individual clicks in a Rasp.

## Buzz
Buzzes are trains of clicks, similar to a Rasp, but so fast that you can't usually distinguish the individual clicks anymore, although this may depend on your spectrogram settings (?). A Buzz occurring after a Rasp or very fast train of echolocation clicks could be an indication of successful foraging. A good rule of thumb is any click train with inter-click-interval (ICI) < 25 ms should be labelled a Buzz.

## Burst Pulse
A burst pulse is when clicks are so close together you cannot distinguish the individual clicks, and it looks tonal.

According to Harald Yurk the sound production mechanism for Burst Pulse is distinct from Rasps and Buzzes, which is why I have nested Burst Pulses under Pulsed Calls (PC) in the diagram above although I'm not sure this is appropriate.

**Comment by oliskir** [comment]
In Beaked Whales, it appears Rasps are distinguished from Buzzes primarily by their different frequency spectrum:

https://onlinelibrary.wiley.com/doi/10.1111/j.1748-7692.2011.00495.x

Rasps and buzzes are said to have similar click repetition rates (median ICI around 5 ms), but Rasp clicks are frequency modulated (FM) similar to regular echolocation clicks whereas Buzz clicks have little or no frequency modulation.

I wonder if something similar holds true for killer whales?

**Comment by oliskir** [comment]
Reply from @scottveirs :  

> Second, in terms of immediate first thoughts:
> 
> 1. RASPS
> 
> I'd never heard the term Rasp used for SRKW signals until the most recent HALLO call. Can you all provide an example (recording of one)? If it's only used by Bigg's KWs, then I suppose discussion of that is better placed in a new repository named `signals-biggs` (my preference if it lives in the Orcasound org) or maybe `signals-TKW` if that's consistent with HALLO nomenclature?
> 
> 2. BUZZ
> 
> I'm leaning towards the idea that an ICI <50ms is a buzz. See this [SRKW ICI Discussion](https://github.com/orcasound/signals-srkw/discussions/14).
> 
> 3. BURST PULSE
> 
> If this sound is different from a `Pulsed Call` (what Orcasound nomenclature simply names a `call`), I'd like to hear a recording. I wonder if it is a redundant term and don't (yet) find it a useful addition to SRKW signal nomenclature.
> 
> 4. FINALLY
> 
> I like your hierarchical presentation and wonder what Orcasound's looks like at the moment. Studying the Google sheet version of a potential shared dictionary, I see something close to this emerging for SRKW signals:
> 
> ```
> signal
> |-- call
> |     |-- discrete
> |     |     |-- S01 (note leading zero!)
> |     |     |-- S02...
> |     |     |-- S42
> |     |-- variable
> |     |     |-- S01?
> |     |     |-- S02? ...
> |     |     |-- S42?
> |     |-- aberrant
> |-- whistle (maybe there should be discrete/variable/aberrant sub-categories?)
> |     |-- SW01
> |     |-- ...
> |     |-- SW07
> |-- click (categories and granularity under discussion!)
> |     |-- pulse (a label for an individual echolocation pulse)
> |     |-- clicks-slow (a sequence label)
> |     |-- clicks-fast (a sequence label)
> |     |-- buzz (a sequence label)
> |     |-- sweep (a sequence label, primarily for waveform annotation, in which received pulse amplitude rises and then falls)
> |-- other
> ```
> 
> I've barely thought about the `other` category (and rarely label any of these sounds), but added to the Gsheet this short list of 6 that might eventually be useful at Orcasound locations (e.g. if we add in-air microphones):
> 
> 1. Percussive
> 2. Cavitation
> 3. Wave
> 4. Blow
> 5. Inhalation
> 6. Raspberry

**Reply by oliskir** [reply]
@scottveirs I will return with some examples of "rasps" and "burst pulses" soon!

**Comment by veirs** [comment]
What is modulated in clicks? Shape? ICI? Other?

**Reply by scottveirs** [reply]
I'll grab the de Soto reference this week from the library, but the open-access abstract -- 

<img src="https://github.com/user-attachments/assets/41d7e4b4-d4c9-4aa2-aea3-392c28f3c10f" width="400">

-- suggests that Blainesville's beaked whales are modulating **frequency** distribution of clicks within a rapid sequence. 

If that's what defines a rasp, @oliskir, we won't likely be able to distinguish them (e.g. from buzzes) in Orcasound data until we deploy hydrophones that are sensitive to much higher frequencies, digitize on the Pisound at 192 or 96kHz (instead of our standard 48kHz), _and_ store the audio data in a lossless format (FLAC)...
